src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -124,6 +130,14 @@ sealed class RsDiagnostic(
             return result.ok()?.value
         }
 
+        private fun ifActualIsStrGetErrTyOfFromStrImplForTy(stringTy: Ty, ty: Ty, items: StdKnownItems, lookup: ImplLookup): Ty? {
+            if (actualTy !is TyReference || actualTy.referenced != TyStr && actualTy.referenced != stringTy) return null
+            val fromStr = items.findFromStrTrait() ?: return null
+            val result = lookup.selectProjectionStrict(TraitRef(ty, BoundElement(fromStr)),
+                fromStr.associatedTypesTransitively.find { it.name == "Err"} ?: return null)
```

Comment:
You can now use `fromStr.findAssociatedType("Err")` instead

Reply:
done